### ISSUE TITLE
flux broker restarts after flux shutdown?

### ISSUE THREAD

**Issue by garlick** [flux-framework/flux-core#5072]
Problem: I ran `flux shutdown --gc` on fluke's rank 0 node, and flux restarted (presumably) automatically 10s later:
```
Apr 06 10:38:57 fluke1 flux[565566]: broker.info[0]: goodbye: goodbye->exit 0.455965ms
Apr 06 10:38:58 fluke1 systemd[1]: flux.service: Main process exited, code=exited, status=42/n/a
Apr 06 10:38:58 fluke1 systemd[1]: flux.service: Failed with result 'exit-code'.
Apr 06 10:39:07 fluke1 systemd[1]: Starting Flux message broker...
Apr 06 10:39:07 fluke1 systemd[1]: Started Flux message broker.
Apr 06 10:39:08 fluke1 flux[1663588]: broker.info[0]: start: none->join 0.10071s
Apr 06 10:39:08 fluke1 flux[1663588]: broker.info[0]: parent-none: join->init 0.08656ms
```
The exit code 42 is supposed to prevent automatic restart per the systemd unit file
```
RestartPreventExitStatus=42
```

Edit: this was flux-core-0.48.0.

Edit: reproduced with a second `flux shutdown`.

**Comment by garlick** [comment]
Still haven't investigated in earnest, but in case the problem is related to a `flux shutdown` leaving the flux unit in failed state -

A possible alternative to RestartPreventExitStatus is to extend the shutdown period using [sd_notify(3)](https://www.freedesktop.org/software/systemd/man/sd_notify.html), discussed [here](https://www.freedesktop.org/software/systemd/man/systemd.service.html#TimeoutStopSec=), and then let flux be shut down using `systemctl stop` without worrying about exceeding TimeoutStopSec while waiting for a long dump or whatever.

**Comment by garlick** [comment]
Some quick test results:

I installed flux-accounting on my test system and was able to reproduce this behavior, where  flux is automatically restarted after a `flux shutdown`.

It seems obvious now, but accounting has `Requires=flux.service` and `Restart=always`.  So when flux is stopped, that causes accounting to stop, then accounting automatically restarts, but it requires flux, so it starts that first.

@grondo noted that here is a `BindTo=` option that can be used in place of `Requires=` which seems to have attractive properties, especially in combination wtih After (which flux accounting also has):

https://www.freedesktop.org/software/systemd/man/systemd.unit.html#BindsTo=

I tried this out and it seemed to resolve the problem
- shutting down flux stops accounting (and both stay down)
- starting flux starts accounting

Perhaps a little more study required but this looks like a promising fix.
